dali/pipeline/executor/executor.cc

```diff
@@ -461,6 +527,34 @@ int Executor<WorkspacePolicy, QueuePolicy>::InferBatchSize(
   return batch_size;
 }
 
+
+template<typename WorkspacePolicy, typename QueuePolicy>
+void Executor<WorkspacePolicy, QueuePolicy>::InitIterationData() {
+  cpu_iteration_id_ = 0;
+  mixed_iteration_id_ = 0;
+  gpu_iteration_id_ = 0;
+  output_iteration_id_ = 0;
+  size_t iteration_data_size = CalcIterationDataSize();
+  iteration_data_.resize(iteration_data_size);
+  for (auto& id : iteration_data_) {
+    id.operator_traces = std::make_shared<operator_trace_map_t>();
+  }
+}
+
+
+template<typename WorkspacePolicy, typename QueuePolicy>
+IterationData &
+Executor<WorkspacePolicy, QueuePolicy>::GetCurrentIterationData(size_t iteration_id) {
+  return iteration_data_[iteration_id % iteration_data_.size()];
+}
+
+
+template<typename WorkspacePolicy, typename QueuePolicy>
+size_t Executor<WorkspacePolicy, QueuePolicy>::CalcIterationDataSize() const {
+  return 2;
```

Comment:
Why 2?

Reply:
It'd be best to explain it on the sketch
```
iteration ID
 CPU 0       1
 MXD   0       1
 GPU       0       1
time 0 1 2 3 4 5 6 7 8 9
           ^^^ Here the iteration 0 is being read by the user (therefore we need to keep it)
               while iteration 1 is being conducted, thus Ops write to OpTraceMap.
```

Reply:
Is such action possible with the `SimpleExecutor`. If it is, than probably this is wrong.